### PR TITLE
refactor/BC-676 Button Componentization

### PR DESCRIPTION
**Pull request by GeoSauer**
This PR adds:

- createButton and createIconButton factory functions to easily create reusable, BC branded, flexible buttons.
- Primary and SecondaryButton with the following props:
    -  colorScheme: defaults to the primary color scheme and can adopt the secondary (red) color scheme
    -  endIcon: adds and styles an icon to the right of the button text
    - startIcon: adds and styles an icon to the left of the button text
    - handler: the function that fires onClick
    - text: text content of the button
        -Other features:
        - accepts Mui Button props, ex: href - renders button as a link, style - allows custom styling object, disabled, etc.
        - automatically sets aria-disabled based on disabled state

-  CreateTaskButton component that matches Figma Hifi 
      - demonstrates flexibility of createButton as it can be called in its own file or directly in the file it's being used and passed props manually. This keeps JSX tidy and allows for use-specific verbose components
      
- IconBtn with the following props:
    - icon: utilizes fetchIcon() to pull from BC specific pool of icons
    - iconSize: choose from small, medium, or large. Matches design spec
    - filled: adds a background 
    - handler: function that fires onClick
        - Other features:
        - accepts Mui Button props, ex: href - renders button as a link, style - allows custom styling object, disabled, etc.
        - automatically sets aria-disabled based on disabled state

- Each component has jsDoc notes that display the accepted props in a tooltip on hover  

- I've also added every icon* in the design spec to the fetchIcon function

Notes:

- there are a handful of icons I couldn't find 1:1 versions of in the Mui library. The should all exist in react icons but fetchIcon will take a bit of tweaking to accommodate. 
- I would like to add the full icon list to the IconBtn jsDoc eventually to make it more dev friendly 
       
[BC-676](https://bootcamper.atlassian.net/browse/BC-676?atlOrigin=eyJpIjoiZWFhZGUyMjU4MWRjNDMxOWE2YzIzOTJlMDFiOGQ0NWIiLCJwIjoiaiJ9)

[BC-676]: https://bootcamper.atlassian.net/browse/BC-676?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/interfaces/components/Button.ts [hunk 1]
```diff
@@ -20,33 +26,80 @@ export interface ConditionalButtonProps extends ButtonProps {
...
-  }
+  [key: string]:
+    | OverridableComponent<SvgIconTypeMap<{}, 'svg'>> & {
+        muiName: string
+      }
 }
 
 export type IconProps = DefaultComponentProps<SvgIconTypeMap>
 
+//commented icons are still needing to be sourced
```
Comment (yeezick) [inline src/interfaces/components/Button.ts:37]: let's remove these dead comments here and add them to the ticket for icons instead to keep this file clean

File: src/utils/components/Icons.tsx [hunk 2]
```diff
@@ -1,21 +1,72 @@
...
+import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank'
+import CheckBoxIcon from '@mui/icons-material/CheckBox'
+import IndeterminateCheckBoxIcon from '@mui/icons-material/IndeterminateCheckBox'
+import RadioButtonUncheckedIcon from '@mui/icons-material/RadioButtonUnchecked'
+import RadioButtonCheckedIcon from '@mui/icons-material/RadioButtonChecked'
+import SearchIcon from '@mui/icons-material/Search'
+//Social
+import InsertLinkOutlinedIcon from '@mui/icons-material/InsertLinkOutlined'
+//TODO: gitHub
+//TODO: linkedIn
```
Comment (yeezick) [inline src/utils/components/Icons.tsx:18]: same here for all these todos, they're better off living in a ticket than in code

**Review by yeezick — CHANGES_REQUESTED** [review]
god this is so clean i'm impressed. 
so far only paginator buttons are implemented correct? we'll need a follow up ticket to implement app-wide? 

waiting on the removal of unneeded comments and points addressed via DM in discord before approving

**Comment by GeoSauer** [comment]
> god this is so clean i'm impressed. so far only paginator buttons are implemented correct? we'll need a follow up ticket to implement app-wide?
> 
> waiting on the removal of unneeded comments and points addressed via DM in discord before approving

Thanks I'm pretty dang happy with it! 
So far nothing is implemented as I wanted to finalize the architecture first. That should be its own ticket spread across several PRs I think.

Comments are removed.
Currently handler is still in place as it is being used where the current Primary/Secondary/Paginator buttons are implemented and will snowball beyond the scope of this ticket if touched. I left a note.

**Review by yeezick — APPROVED** [review]

**Pull request merged**